plugin/path_role_set.go

```diff
@@ -6,123 +6,113 @@ import (
+				Type:        framework.TypeString,
+				Description: fmt.Sprintf("Type of secret generated for this role set. Defaults to '%s'", SecretTypeAccessToken),
+				Default:     SecretTypeAccessToken,
 			},
-			HelpSynopsis:    pathRoleSetHelpSyn,
-			HelpDescription: pathRoleSetHelpDesc,
-		},
-		// Path to rotate role set service accounts
-		{
-			Pattern: fmt.Sprintf("roleset/%s/rotate", framework.GenericNameRegex("name")),
-			Fields: map[string]*framework.FieldSchema{
-				"name": {
-					Type:        framework.TypeString,
-					Description: "Name of the role.",
-				},
+			"project": {
+				Type:        framework.TypeString,
+				Description: "Name of the GCP project that this roleset's service account will belong to.",
+			},
+			"bindings": {
+				Type:        framework.TypeString,
+				Description: "Bindings configuration string.",
 			},
-			ExistenceCheck: b.pathRoleSetExistenceCheck,
-			Operations: map[logical.Operation]framework.OperationHandler{
-				logical.UpdateOperation: &framework.PathOperation{
-					Callback: b.pathRoleSetRotateAccount,
-				},
+			"token_scopes": {
+				Type:        framework.TypeCommaStringSlice,
+				Description: `List of OAuth scopes to assign to credentials generated under this role set`,
 			},
-			HelpSynopsis:    pathRoleSetRotateHelpSyn,
-			HelpDescription: pathRoleSetRotateHelpDesc,
 		},
-		// Path to rotating role set service account key used to generate access tokens
-		{
-			Pattern: fmt.Sprintf("roleset/%s/rotate-key", framework.GenericNameRegex("name")),
-			Fields: map[string]*framework.FieldSchema{
-				"name": {
-					Type:        framework.TypeString,
-					Description: "Name of the role.",
-				},
+		ExistenceCheck: b.pathRoleSetExistenceCheck,
+		Operations: map[logical.Operation]framework.OperationHandler{
+			logical.DeleteOperation: &framework.PathOperation{
+				Callback: b.pathRoleSetDelete,
+			},
+			logical.ReadOperation: &framework.PathOperation{
+				Callback: b.pathRoleSetRead,
+			},
+			logical.CreateOperation: &framework.PathOperation{
+				Callback: b.pathRoleSetCreateUpdate,
 			},
-			ExistenceCheck: b.pathRoleSetExistenceCheck,
-			Operations: map[logical.Operation]framework.OperationHandler{
-				logical.UpdateOperation: &framework.PathOperation{
-					Callback: b.pathRoleSetRotateKey,
-				},
+			logical.UpdateOperation: &framework.PathOperation{
+				Callback: b.pathRoleSetCreateUpdate,
 			},
-			HelpSynopsis:    pathRoleSetRotateKeyHelpSyn,
-			HelpDescription: pathRoleSetRotateKeyHelpDesc,
 		},
-		// Paths for listing role sets
-		{
-			Pattern: "rolesets/?",
-			Operations: map[logical.Operation]framework.OperationHandler{
-				logical.ListOperation: &framework.PathOperation{
-					Callback: b.pathRoleSetList,
-				},
+		HelpSynopsis:    pathRoleSetHelpSyn,
+		HelpDescription: pathRoleSetHelpDesc,
+	}
+}
+
+func pathRoleSetList(b *backend) *framework.Path {
+	// Paths for listing role sets
+	return &framework.Path{
+		Pattern: "rolesets?/?",
```

Comment:
Is the `?` after `rolesets` intended? `rolesets?` seems weird

Reply:
yup - we had two paths for roleset/? and rolesets/? so might as well match 0-1 of the s

Reply:
Ack